src/network/core/game_info.cpp

```diff
@@ -106,6 +106,22 @@ bool IsNetworkCompatibleVersion(const char *other)
 	return hash1 && hash2 && (strncmp(hash1, hash2, GITHASH_SUFFIX_LEN) == 0);
 }
 
+/**
+ * Check if an game entry is compatible with our client.
+ */
+void CheckGameCompatability(NetworkGameInfo &ngi)
+{
+	/* Check if we have all the GRFs on the client-system too. */
+	ngi.compatible = true;
+	for (const GRFConfig *c = ngi.grfconfig; c != nullptr; c = c->next) {
+		if (c->status == GCS_NOT_FOUND) ngi.compatible = false;
+	}
+
+	/* Check if we are allowed on this server based on the revision-match */
+	ngi.version_compatible = IsNetworkCompatibleVersion(ngi.server_revision);
```

Comment:
I would move this code to the begin of the function, and set ngi.compatible to ngi.version_compatible. Then the if below is not needed anymore.